File: pgpainless-core/src/main/java/org/pgpainless/key/protection/UnprotectedKeysProtector.java
// SPDX-FileCopyrightText: 2018 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0

package org.pgpainless.key.protection;

import javax.annotation.Nullable;

import org.bouncycastle.openpgp.operator.PBESecretKeyDecryptor;
import org.bouncycastle.openpgp.operator.PBESecretKeyEncryptor;

/**
 * Implementation of the {@link SecretKeyRingProtector} which assumes that all handled keys are not password protected.
 */
public class UnprotectedKeysProtector implements SecretKeyRingProtector {

    @Override
    public boolean hasPassphraseFor(long keyId) {
        return true;
    }

    @Override
    @Nullable
    public PBESecretKeyDecryptor getDecryptor(long keyId) {
        return null;
    }

    @Override
    @Nullable
    public PBESecretKeyEncryptor getEncryptor(long keyId) {
        return null;
    }
}

File: pgpainless-core/src/main/kotlin/org/pgpainless/key/protection/UnprotectedKeysProtector.kt
// SPDX-FileCopyrightText: 2023 Paul Schaub <vanitasvitae@fsfe.org>
//
// SPDX-License-Identifier: Apache-2.0
package org.pgpainless.key.protection

import org.bouncycastle.bcpg.KeyIdentifier
import org.bouncycastle.openpgp.api.OpenPGPKey
import org.bouncycastle.openpgp.operator.PBESecretKeyDecryptor
import org.bouncycastle.openpgp.operator.PBESecretKeyEncryptor

/**
 * Implementation of the [SecretKeyRingProtector] which assumes that all handled keys are not
 * password protected.
 */
class UnprotectedKeysProtector : SecretKeyRingProtector {
    override fun hasPassphraseFor(keyIdentifier: KeyIdentifier): Boolean = true

    override fun getDecryptor(keyIdentifier: KeyIdentifier): PBESecretKeyDecryptor? = null

    override fun getEncryptor(keyIdentifier: KeyIdentifier): PBESecretKeyEncryptor? = null

    override fun getKeyPassword(p0: OpenPGPKey.OpenPGPSecretKey?): CharArray? = null
}
